client/scripts/views/pages/subscriptions/notificationSettings.ts

```diff
@@ -89,25 +89,22 @@ interface IBatchedSubscriptionRowAttrs {
 }
 
 interface IBatchedSubscriptionRowState {
-  subscriptions: NotificationSubscription[];
-  paused: boolean;
   option: string;
 }
 
 const BatchedSubscriptionRow: m.Component<IBatchedSubscriptionRowAttrs, IBatchedSubscriptionRowState> = {
-  oninit: (vnode) => {
-    vnode.state.subscriptions = vnode.attrs.subscriptions;
-  },
   view: (vnode) => {
-    const { label, bold } = vnode.attrs;
-    const { subscriptions } = vnode.state;
+    const { label, bold, subscriptions } = vnode.attrs;
     const someActive = subscriptions.some((s) => s.isActive);
     const everyActive = subscriptions.every((s) => s.isActive);
     const someEmail = subscriptions.some((s) => s.immediateEmail);
-    if (everyActive && someEmail) {
+    const everyEmail = subscriptions.some((s) => s.immediateEmail);
+    if (everyActive && everyEmail) {
       vnode.state.option = 'Notifications on (app + email)';
-    } else if (everyActive) {
+    } else if (everyActive && !someEmail) {
       vnode.state.option = 'Notifications on (app only)';
+    } else if (someActive) {
+      vnode.state.option = 'Some notifications on';
```

Comment:
The `New threads (all communities)` option should show "Some notifications on" if the new threads option is only selected for some communities, right?

Reply:
Correct. It is the "indeterminate" state for the dropdown menu. It's not an option to select, but a default value for the button trigger to the menu.